src/AnticheatMgr.cpp

```diff
@@ -127,8 +132,64 @@ void AnticheatMgr::StartHackDetection(Player* player, MovementInfo movementInfo,
             BGStartExploit(player, movementInfo);
         }
     }
-    m_Players[key].SetLastMovementInfo(movementInfo);
-    m_Players[key].SetLastOpcode(opcode);
+    m_Players[key].SetLastInformations(movementInfo, opcode, GetPlayerCurrentSpeedRate(player));
+}
+
+uint32 AnticheatMgr::GetTeleportSkillCooldownDurationInMS(Player* player) const
+{
+    switch (player->getClass())
+    {
+        case CLASS_ROGUE:
+            if (player->HasAura(FILTHY_TRICKS_RANK_2))
+                return 20000u;
+            else if (player->HasAura(FILTHY_TRICKS_RANK_1))
+                return 25000u;
+            else
+                return 30000u;
+        case CLASS_MAGE:
+            if (player->HasAura(BLINK_COOLDOWN_REDUCTION)) // Bonus from Vanilla/Early TBC pvp gear.
+                return 13000u;
+            else
+                return 15000u;
+        default:
+            return 0u;
```

Comment:
```suggestion
    switch (player->getClass())
    {
        case CLASS_ROGUE:
            if (player->HasAura(FILTHY_TRICKS_RANK_2))
                return 20000u;
            else if (player->HasAura(FILTHY_TRICKS_RANK_1))
                return 25000u;
            return 30000u;
        case CLASS_MAGE:
            if (player->HasAura(BLINK_COOLDOWN_REDUCTION)) // Bonus from Vanilla/Early TBC pvp gear.
                return 13000u;
            return 15000u;
        default:
            return 0u;
```

You don't need the elses here as well